pkg/controller/keyvault/gateway_kv_serviceaccount.go

```diff
@@ -0,0 +1,137 @@
+package keyvault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/controllername"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/metrics"
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+var kvSaControllerName = controllername.New("gateway", "keyvault", "serviceaccount")
+
+// KvServiceAccountReconciler reconciles the default "azure-app-routing-kv" ServiceAccount in each namespace where users
```

Comment:
can users federate the credential with a subject of a non existing service account and have this work?

1. user federate non existing credential in namespace
2. user creates gateway object and this controller makes service account

does the federation properly apply to the newly created SA? Does it apply instantly or is there a wait time?

Reply:
I'm not a huge fan of the pattern of passing in client id through gateway and using that for the SA. It's a lot of indirection. Why doesn't user just create SA and pass that in instead? Drawing the line at us making SA but still taking in client id makes the entire process convoluted because gateway objects are many to one for namespaces and you can also have many tls configs in a single gateway object

Reply:
I can vouch that you can create federated credentials without having a SA present in the cluster.

Reply:
Totally forgot to follow up on this but yes - confirmed via manual testing. It uses the oidc URL to specify the cluster, and then from what I can tell there's no validation against the ns or SA that's specified - it seems like portal just takes care of formatting the string for the oidc cred